ocaml/xapi/repository.ml

```diff
@@ -0,0 +1,127 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+module Unixext = Xapi_stdext_unix.Unixext
+
+module D = Debug.Make (struct let name = "repository" end)
+
+open D
+
+let cmd_yum = "/usr/bin/yum"
+let cmd_rm = "/usr/bin/rm"
```

Comment:
Don't use `rm` through the shell, but `Unix.unlink`.

Reply:
It is used to remove a non-empty directory.
It looks like "Unix.unlink" works for a file only.  And Unix. rmdir works for an empty directory.
May I please know any existing utility before writing one?

Reply:
If it needs a recursive delete, then you could copy the function from https://github.com/xapi-project/xenopsd/blob/master/lib/xenops_utils.ml#L186-L203.